forgerock-auth/src/main/java/org/forgerock/android/auth/devicebind/BiometricAuthenticator.kt

```diff
@@ -0,0 +1,116 @@
+     * @param context Application Context
+     * @return statusResult Listener for receiving Biometric changes
+     */
+    override suspend fun authenticate(context: Context): DeviceBindingStatus =
+        withContext(Dispatchers.Main) {
+            suspendCoroutine { continuation ->
+                //The keys may be removed due to pin change
+                val privateKey = cryptoKey.getPrivateKey()
+                if (privateKey == null) {
+                    continuation.resume(DeviceBindingErrorStatus.ClientNotRegistered())
+                } else {
+                    val listener = object : BiometricPrompt.AuthenticationCallback() {
+
+                        override fun onAuthenticationError(errorCode: Int,
+                                                           errString: CharSequence) {
+                            when (errorCode) {
+                                BiometricPrompt.ERROR_CANCELED, BiometricPrompt.ERROR_USER_CANCELED, BiometricPrompt.ERROR_NEGATIVE_BUTTON -> continuation.resume(
+                                    DeviceBindingErrorStatus.Abort(errString.toString(),
+                                        code = errorCode))
+
+                                BiometricPrompt.ERROR_TIMEOUT -> continuation.resume(
+                                    DeviceBindingErrorStatus.Timeout(
+                                        errString.toString(),
+                                        code = errorCode))
+
+                                BiometricPrompt.ERROR_NO_BIOMETRICS, BiometricPrompt.ERROR_NO_DEVICE_CREDENTIAL, BiometricPrompt.ERROR_HW_NOT_PRESENT -> continuation.resume(
+                                    DeviceBindingErrorStatus.Unsupported(errString.toString(),
+                                        code = errorCode))
+
+                                BiometricPrompt.ERROR_VENDOR -> continuation.resume(
+                                    DeviceBindingErrorStatus.Unsupported(
+                                        errString.toString(),
+                                        code = errorCode))
+
+                                BiometricPrompt.ERROR_LOCKOUT_PERMANENT, BiometricPrompt.ERROR_LOCKOUT, BiometricPrompt.ERROR_NO_SPACE, BiometricPrompt.ERROR_HW_UNAVAILABLE, BiometricPrompt.ERROR_UNABLE_TO_PROCESS -> continuation.resume(
+                                    DeviceBindingErrorStatus.UnAuthorize(errString.toString(),
+                                        code = errorCode))
+
+                                else -> {
+                                    continuation.resume(DeviceBindingErrorStatus.Unknown(errString.toString(),
+                                        code = errorCode))
+                                }
+                            }
+                        }
+
+                        override fun onAuthenticationSucceeded(result: BiometricPrompt.AuthenticationResult) {
+                            result.cryptoObject?.let {
```

Comment:
i think we can do one line 

  result.cryptoObject?.signature?.let {
                                continuation.resume(Success(privateKey, it))
                                return
                            }

Reply:
Updated